### ISSUE TITLE
Style/NumericPredicates is incompatible with RSpec matchers

### ISSUE THREAD

**Issue by ragesoss** [bbatsov/rubocop#3444]
## Expected behavior

Rubocop should not warn about or autocorrect RSpec comparison matchers `<` or `>` for replacement with `.positive?` and `.negative?`, because the latter are undefined methods for RSpec matchers. 

See https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers

## Actual behavior

With TargetRubyVersion 2.3, autocorrect will replace comparisons in specs, breaking them. This behavior occurs even with the `-R` flag and if rubocop-rspec is in use.

## Steps to reproduce the problem
Create an spec like this:

```ruby
describe 'rspec matchers' do
  it 'autocorrects badly' do
    a = 1
    expect(a).to be > 0
  end
end
```

In `.rubocop.yml`, include
```
AllCops:
  TargetRubyVersion: 2.3
```

## RuboCop version
```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-linux)
```

**Comment by bbatsov** [comment]
This can't really be fixed. Just exclude your spec dir for this cop in your `.rubocop.yml`.

**Comment by ragesoss** [comment]
It's easy to work around, of course, but it's a dangerous default for rails / rspec projects. It would be nice if, for example, using rubocop-rspec would disable that by default for spec files.
